backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/cardlist/stack/internal/BpkStackCardListImpl.kt

```diff
@@ -0,0 +1,136 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import net.skyscanner.backpack.compose.button.BpkButton
+import net.skyscanner.backpack.compose.button.BpkButtonIconPosition
+import net.skyscanner.backpack.compose.button.BpkButtonType
+import net.skyscanner.backpack.compose.cardlist.stack.BpkStackCardAccessoryStyle
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.sectionheader.BpkSectionHeader
+import net.skyscanner.backpack.compose.sectionheader.BpkSectionHeaderButton
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.ChevronDown
+import net.skyscanner.backpack.compose.tokens.ChevronUp
+
+@Composable
+internal fun BpkStackCardListImpl(
+    title: String,
+    description: String?,
+    totalCount: Int,
+    modifier: Modifier = Modifier,
+    accessoryStyle: BpkStackCardAccessoryStyle? = null,
+    headerButton: BpkSectionHeaderButton? = null,
+    accessibilityHeaderTagEnabled: Boolean? = true,
+    content: @Composable ((Int) -> Unit),
+) {
+
+    var isExpanded by rememberSaveable { mutableStateOf(false) }
+    val targetCount = when (accessoryStyle) {
+        is BpkStackCardAccessoryStyle.Expand -> {
+            if (accessoryStyle.expandedCount < 0 || accessoryStyle.collapsedCount < 0) {
+                throw IllegalArgumentException("expandedCount/collapsedCount can not be negative")
+            }
+            if (accessoryStyle.expandedCount < accessoryStyle.collapsedCount) {
+                throw IllegalArgumentException("expandedCount should be greater than collapsedCount")
+            }
+            if (accessoryStyle.expandedCount <= MINIMUM_EXPANDED_COUNT) {
+                totalCount
+            } else {
+                if (isExpanded) accessoryStyle.expandedCount else accessoryStyle.collapsedCount
+            }
+        }
+        else -> totalCount
+    }
+
+    LazyColumn(
+        modifier = modifier,
+        contentPadding = PaddingValues(start = BpkSpacing.Base, end = BpkSpacing.Base),
+        verticalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+    ) {
+        item {
+            BpkSectionHeader(
+                title = title,
+                description = description,
+                modifier = Modifier,
+                button = headerButton,
+                accessibilityHeaderTagEnabled = accessibilityHeaderTagEnabled,
+            )
+        }
+
+        items(targetCount) {
+            Box(Modifier.animateItem()) {
+                content(it)
+            }
+        }
+
+        item {
+            accessoryStyle?.let {
+                when (it) {
+                    is BpkStackCardAccessoryStyle.Expand -> {
+                        if (it.expandedCount > MINIMUM_EXPANDED_COUNT) {
```

Comment:
We shouldn't be using `MINIMUM_EXPANDED_COUNT` for this check IMO. This would result in always showing the accessory when `expandedCount  > 2`, but if `collapsedCount` AND `expandedCount` are both equal to 4, for example, we wouldn't want to show the accessory in this case. So I think we could be checking against `collapsedCount` instead:
```suggestion
                        if (it.expandedCount > it.collapsedCount) {
```

Reply:
If both `expandedCount ` and `collapsedCount ` are equal, it means the user intentionally wants to hide the expand accessory. For this use-case the user should set the expand accessory to null as it is optional.

Reply:
Same as [my reply above](https://github.com/Skyscanner/backpack-android/pull/2257#discussion_r2042268030), this makes sense now 🙏